constraint/pkg/client/regolib/src.go

```diff
@@ -4,10 +4,24 @@ const (
 	targetLibSrc = `
 package hooks["{{.Target}}"]
 
+# Skip constraint evaluation for a given target if constraint has namespaceSelector but Namespace is not cached
```

Comment:
k8s-specific code should not be in the framework. The framework should provide a way for the target-handler to auto-reject and put the NS selector auto-rejection in the targethandler

Reply:
How about adding a new rule similar to `matching_constraints[constraint]` in `Library()` ? For example, the new rule could be called ` autoreject_constraints[constraint, review]` and implemented by the target handler.

Reply:
Per our conv. on Friday, that SGTM